k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/ImapStore.java

```diff
@@ -488,4 +533,9 @@ public void setCombinedPrefix(String prefix) {
             combinedPrefix = prefix;
         }
     }
+
+    private boolean supportsListExtended(ImapConnection connection) {
+        return connection.hasCapability(Capabilities.SPECIAL_USE)
```

Comment:
does this depend on the service provider?

Reply:
yes it does, and it is tricky. For example Microsoft accounts do not offer this. This means they don't claim that the Folder flags are 100% sure identification of folders identity/role. But we use the flags anyway if we find them.